dxp/resource_serving.md

```diff
@@ -0,0 +1,161 @@
+
+When a cache hit occurs, the server returns the same response it sent last time (including headers, not just the content).
+
+### GZip Filter
+
+This filter compresses the response with the `gzip` algorithm. Its activation can be altered:
+
+1. Request parameter `compress`: when set to `false` filter is inhibited.
+2. The browser's `Accept-Encoding` header: unless `gzip` is accepted, the filter is disabled.
+
+### Language Filter
+
+This filter first determines the request's locale from a variety of sources: the `languageId` parameter, the `ThemeDisplay`, `doAsUserLanguageId` parameter, stored session value, user, cookies, ... See [the class' source code](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/servlet/filters/language/LanguageFilter.java) for more detail.
+
+Once the locale is determined, it replaces each occurrence of `Liferay.Language.get(...)` by the translated value according to the deployed `Language.properties` files.
+
+> This filter is the reason why placing any `Liferay.Language.get(...)` call in your JavaScript source works synchronously without calling the server again or executing any code.
+
+> And of course, it is also the reason why dynamic language keys cannot be used in JavaScript.
+
+### Aggregate Filter
+
+This filter is used to bundle multiple JavaScript into a minified single resource (called bundle) that can be fetched from the server, as well as to minify single JavaScript or CSS files.
+
+#### JavaScript Bundle
+
+This filter was used to serve _JavaScript bundles_ (sets of `.js` files that are bundled into a single resource) which were identified in the `portal.properties` file with the `javascript.bundle.ids` property. Out of the box, DXP came configured with two bundles: `javascript.barebone.files` and `javascript.everything.files`.
+
+Each bundle had a location in the server's URL space defined by the portal property `javascript.bundle.dir`. So, for example, both `javascript.bundle.dir[javascript.barebone.files]` and `javascript.bundle.dir[javascript.everything.files]` were located in the root of the server (`/`). This is the place from which the files in the bundle are retrieved when requested.
+
+To finish with, any `.js` resource you request (no matter what the URL is) which has valid `minifierType` and `minifierBundleId`/`bundleId` request parameters returns a cached copy of the JavaScript bundle aggregate. Any value is valid for `minifierType` while only defined bundles can be passed as `minifierBundleId`/`bundleId`.
+
+However, as it is now, DXP doesn't define any file in `portal.properties` for any of the two predefined bundles because they were migrated to a `DynamicInclude` service registration. Also, the mechanism is deprecated, though it still works if someone configures it.
```

Comment:
I think you should actually lead with this at the top. eg. first sentence under the heading would be:

> This is a deprecated and unused mechanism, but it still works if someone configures it.